index.js

```diff
@@ -1,106 +1,129 @@
  */
-var usersToDeprovision = function(csv) {
-    var oneMonthAgo = moment().subtract(1, "months");
-
-    return csv.then(function(data) {
-        var realUsers = data.filter(function(row) {
-            return row["Email Address"] !== "";
-        });
-
-        var guests = realUsers.filter(function(row) {
-            return row["Internal"] === "false";
-        });
-
-        var rowsToDeprovision = guests.filter(function(row) {
-            var lastActivityMoment = moment(row["Last Activity"]);
-
-            if (!lastActivityMoment.isValid()) {
-                // Probably hasn't logged in, fall back to invitation date instead
-                lastActivityMoment = moment(row["Date Joined Organization"]);
-            }
-            return lastActivityMoment.isBefore(oneMonthAgo);
-        });
-
-        return rowsToDeprovision.map(function(row) {
-            return row["Email Address"];
-        });
+const usersToDeprovision = csv => {
+  const threshold = moment().subtract(parseInt(process.argv[5], 10), "days");
+
+  return csv.then(function(data) {
+    const realUsers = data.filter(row => {
+      return row["Email Address"] !== "";
     });
-};
 
-var main = function() {
-    if (process.argv.length < 5 || process.argv.length > 6) {
-        console.log("Usage:");
-        console.log("node index.js <service account token> <member export csv url> <organization id> [action]");
-    }
+    const guests = realUsers.filter(row => {
+      return row["Internal"] === "false";
+    });
 
-    if (process.argv[5] !== "action") {
-        console.log("In dry-run mode. Add 'action' to the command line to actually deprovision users");
-    }
+    const rowsToDeprovision = guests.filter(row => {
+      const lastActivityMoment = moment(row["Last Activity"]);
+
+      if (!lastActivityMoment.isValid()) {
+        // Probably hasn't logged in, fall back to invitation date instead
+        lastActivityMoment = moment(row["Date Joined Organization"]);
+      }
+      return lastActivityMoment.isBefore(threshold);
+    });
 
-    var deprovision = deprovisionFunc();
-
-    var csvData = loadRemoteCsv(process.argv[3]);
-    usersToDeprovision(csvData).then(function (emails) {
-        emails.forEach(function (email) {
-            if (process.argv[5] === "action") {
-                console.log("Deprovisioning", email);
-                deprovision(email);
-            } else {
-                console.log("Planning to deprovision", email);
-            }
-        })
+    return rowsToDeprovision.map(row => {
+      return row["Email Address"];
     });
+  });
+};
+
+const main = () => {
+  if (process.argv.length < 6 || process.argv.length > 7) {
+    console.log("Usage:");
+    console.log(
+      "node index.js <service account token> <member export csv path or url> <organization id> <# of days inactive threshold> [action]"
+    );
+  }
+
+  if (process.argv[6] !== "action") {
```

Comment:
I regret not using a command line options module now. This edit will break existing command lines, because the number of days is mandatory, and appears before the "action", so it'll try to use "action" as the number of days.

If you feel like pulling off the band aid and switching to --domain_id= --action etc, I'd support that, and we should do it now, before changing the command line format multiple times.